File: pronacej-backend-seguridad/src/main/java/net/latinus/sistema/integral/gestion/seguridad/service/param/AlertaService.java
package net.latinus.sistema.integral.gestion.seguridad.service.param;

import jakarta.servlet.http.HttpServletRequest;
import net.latinus.sistema.integral.gestion.seguridad.model.both.AlertaDTO;
import net.latinus.sistema.integral.gestion.seguridad.model.both.BodyEncriptado;
import net.latinus.sistema.integral.gestion.seguridad.model.response.PaginacionResponse;
import net.latinus.sistema.integral.gestion.seguridad.model.response.RespuestaPorDefectoAuditoria;

import java.util.List;

public interface AlertaService {

    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data AlertaDTO
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado     request peticion.
     * @return RespuestaPorDefectoAuditoria<EncuestaDTO>
     */
    RespuestaPorDefectoAuditoria<PaginacionResponse<AlertaDTO>> obtenerListaAlertas(HttpServletRequest httpServletRequest,
                                                                                    BodyEncriptado bodyEncriptado);

    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data AlertaDTO
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado     request peticion.
     * @return RespuestaPorDefectoAuditoria<EncuestaDTO>
     */
    RespuestaPorDefectoAuditoria<List<AlertaDTO>> obtenerAlertas(HttpServletRequest httpServletRequest,
                                                                 BodyEncriptado bodyEncriptado);

    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data Boolean si la
     * encuesta se creo con exito
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado     objeto encuesta dto.
     * @return RespuestaPorDefectoAuditoria<Boolean>
     */
    RespuestaPorDefectoAuditoria<Boolean> crearAlerta(HttpServletRequest httpServletRequest,
                                                      BodyEncriptado bodyEncriptado);

    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data Boolean si la
     * encuesta se creo con exito
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado     objeto encuesta dto.
     * @return RespuestaPorDefectoAuditoria<Boolean>
     */
    RespuestaPorDefectoAuditoria<Boolean> actualizarAlerta(HttpServletRequest httpServletRequest,
                                                           BodyEncriptado bodyEncriptado);

    /**
     * Devuelve un objeto RespuestaPorDefectoAuditoria con data Boolean si la
     * encuesta se creo con exito
     *
     * @param httpServletRequest request peticion.
     * @param bodyEncriptado     objeto encuesta dto.
     * @return RespuestaPorDefectoAuditoria<Boolean>
     */
    RespuestaPorDefectoAuditoria<Boolean> removerAlerta(HttpServletRequest httpServletRequest,
                                                        BodyEncriptado bodyEncriptado);
}

File: pronacej-front/src/app/modules/general/services/alerta.service.ts
import { Injectable } from '@angular/core';
import { AlertaDTO } from 'app/core/model/both/AlertaDTO.model';
import { PaginacionRequest } from 'app/core/model/request/PaginacionRequest.model';
import { PaginacionResponse } from 'app/core/model/response/PaginacionResponse.model';
import { RespuestaPorDefecto } from 'app/core/model/response/RespuestaPorDefecto.model';
import { BackendService } from 'app/core/services/backend.service';
import { Observable } from 'rxjs';

@Injectable({
    providedIn: 'root',
})
export class AlertaService {
    private path = '/alerta';

    constructor(private backendService: BackendService) {}

    obtenerListaAlertas(
        paginacionRequest: PaginacionRequest,
        nemonicoMenu: string
    ): Observable<RespuestaPorDefecto<PaginacionResponse<AlertaDTO>>> {
        let endPoint = this.path + '/obtenerListaAlertas';
        return this.backendService.postFinal(
            endPoint,
            paginacionRequest,
            nemonicoMenu
        );
    }

    obtenerAlertas(
        alertaDTO: AlertaDTO,
        nemonicoMenu: string
    ): Observable<RespuestaPorDefecto<AlertaDTO[]>> {
        let endPoint = this.path + '/obtenerAlertas';
        return this.backendService.postFinal(endPoint, alertaDTO, nemonicoMenu);
    }

    crearAlerta(
        alertaDTO: AlertaDTO,
        nemonicoMenu: string
    ): Observable<RespuestaPorDefecto<Boolean>> {
        let endPoint = this.path + '/crearAlerta';
        return this.backendService.postFinal(endPoint, alertaDTO, nemonicoMenu);
    }

    actualizarAlerta(
        alertaDTO: AlertaDTO,
        nemonicoMenu: string
    ): Observable<RespuestaPorDefecto<Boolean>> {
        let endPoint = this.path + '/actualizarAlerta';
        return this.backendService.postFinal(endPoint, alertaDTO, nemonicoMenu);
    }

    removerAlerta(
        alertaDTO: AlertaDTO,
        nemonicoMenu: string
    ): Observable<RespuestaPorDefecto<Boolean>> {
        let endPoint = this.path + '/removerAlerta';
        return this.backendService.postFinal(endPoint, alertaDTO, nemonicoMenu);
    }
}
